### DISCUSSION TITLE
FrameBuffer.rect Bug?

### DISCUSSION THREAD

**Discussion by webdeb** [micropython/micropython#10418] [General]
I am writing a simple menu renderer, where you can provide an active item.
To make the item visible I wanted to create a white react and overlay the text in black.
(My expectation is that this will work) However, when the `rect` method throws an interesting exception, which says, that I am using 7 args instead of 6. This is very confusing.. See the code and the error below:

I am using ssd1306 if that matters.. 

```py
def render_menu(items, active):
  menubuf = framebuf.FrameBuffer(bytearray(1024), 128, 64, framebuf.MONO_HLSB)
  menubuf.text("Menu", 0, 0, 1)
  menubuf.hline(0, 12, 128, 1)

  for idx, item in enumerate(items, start=1):
    if active == item:
      text_y_pos = 8 + 12 * idx
      menubuf.rect(0, text_y_pos - 2, 128, 12, 1, 0)
      menubuf.text(item, 2, text_y_pos, 0)
    else:
      menubuf.text(item, 2, 8 + 12 * idx, 1)

  display.blit(menubuf, 0, 0)
  display.show()
```

<img width="859" alt="Bildschirm­foto 2023-01-05 um 20 25 49" src="https://user-images.githubusercontent.com/14992140/210863838-3bdb80a7-aed4-435f-8587-84d5453cf27a.png">

**Comment by robert-hh** [comment]
The 6th argument for fill was added in August 2022. So if your firmware is older, the last argument must be omitted. Alternatively just use the most recent daily build. The number  of arguments in the error message always include the (invisible) instance. So if you supply 6 arguments in Python, the C code driver gets 7.

**Reply by webdeb** [reply]
Oh, thank you. Do you think I can just copy the file from the actual master?
https://github.com/micropython/micropython/blob/bf49a087b2005e45f7ec5672f5ecd2668e2deb4f/extmod/modframebuf.c

**Reply by webdeb** [reply]
sure, compiling the port etc.. but is there anything else?

**Reply by robert-hh** [reply]
Do not copy a single file. Use the actual state of the repository. If you just used standard firmware, get the actual firmware from the download site.

You can simple omit the last argument in your code. It allows to paint a  filled rectangle instead of just the outline.

**Reply by webdeb** [reply]
oh yeah.. read the docs they said..

<img width="736" alt="Bildschirm­foto 2023-01-05 um 21 36 45" src="https://user-images.githubusercontent.com/14992140/210875232-245e3101-7000-41da-9b1f-10926943ea5e.png">

In the previous versions its `fill_rect`

**Reply by Rumidom** [reply]
i got this error on MicroPython v1.22.2 on 2024-02-22, has something changed? doc says to add 6th argument for fill
